youtube_dl/extractor/safari.py

```diff
@@ -38,7 +38,8 @@ def _login(self):
             'Downloading login page')
 
         def is_logged(urlh):
-            return 'learning.oreilly.com/home/' in urlh.geturl()
+            return ('https://learning.oreilly.com/member/login/' == urlh.geturl()
+                    or 'learning.oreilly.com/home/' in urlh.geturl())
```

Comment:
This would be more in keeping with the previous code:
```suggestion
            return any('learning.oreilly.com/%s/' % x in urlh.geturl() for x in ('member/login', 'home'))
```

Reply:
Depending on how oreilly is accessed, the URL can end up being "https://learning.oreilly.com/member/login/?next=[snip]" for cases where the user is not yet logged in.  So we don't want to catch that case here with the `in` instead of `==`.

Given that, is there a more conventional way to format the check being done here?

Reply:
Maybe 
```sugg
            return bool(re.search(r'learning.oreilly.com/(?:%s)' % '|'.join(('member/login/$', 'home/')), urlh.geturl()))
```
or 
```sugg
            parsed_url = compat_urlparse.urlparse(url)
            return parsed_url.hostname.endswith('learning.oreilly.com') and (
                parsed_url.path.startswith('/home/')
                or (parsed_url.path == '/member/login' and not parsed_url.query))
```
Pick either of these or your original, check your test and then I think it's good to go.